test/functional/load_esnext.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const version = process.version;
+
+function loadTests() {
+  const directory = path.resolve.apply(path, arguments);
+  fs
+    .readdirSync(directory)
+    .filter(filePath => filePath.match(/.*\.js$/))
+    .map(filePath => path.resolve(directory, filePath))
+    .forEach(x => require(x));
+}
+
+let supportES2017 = false;
+try {
+  new Function('return (async function foo() {})();')();
+  supportES2017 = true;
+} catch (e) {
+  supportES2017 = false;
+}
+
+const supportES2018 = !!Symbol.asyncIterator;
+
+if (supportES2017) {
```

Comment:
I know in years past the driver has opted to simply omit results for tests which are skipped, but isn't it preferable to use `this.skip()` to actually show where this is happening?

Reply:
Unfortunately, these tests are dynamically required specifically b/c attempting to parse them with a lower Node version will instantly break the test runner. For example, if Node 4 attempts to parse the following:

```js
function foo() {
  return bar;

  async function bar() {
    const x = await 42;
    return x;
  }
}
```

it will error. That is why these need to be in separate files.

Reply:
yeah I meant more like:

```
describe('blah', function() {
   it('halb', function() {
     if (!supportES2017) {
       this.skip();
     }

     loadTests(...);
   });
});
```

or something along those lines